packages/adapter-d1/src/index.ts

```diff
@@ -0,0 +1,353 @@
+export const GET_SESSION_BY_TOKEN_SQL = `
+  SELECT id, sessionToken, userId, expires
+  FROM authjs_sessions
+  WHERE sessionToken = ?`;
+export const UPDATE_SESSION_BY_SESSION_TOKEN_SQL = `UPDATE authjs_sessions SET expires = ? WHERE sessionToken = ?`;
+export const DELETE_SESSION_SQL = `DELETE FROM authjs_sessions WHERE sessionToken = ?`;
+export const DELETE_SESSION_BY_USER_ID_SQL = `DELETE FROM authjs_sessions WHERE userId = ?`;
+
+// ACCOUNT
+export const CREATE_ACCOUNT_SQL = `
+  INSERT INTO authjs_accounts (
+    id, userId, type, provider, 
+    providerAccountId, refresh_token, access_token, 
+    expires_at, token_type, scope, id_token, session_state,
+    oauth_token, oauth_token_secret
+  ) 
+  VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?)`;
+export const GET_ACCOUNT_BY_ID_SQL = `SELECT * FROM authjs_accounts WHERE id = ? `
+export const GET_ACCOUNT_BY_PROVIDER_AND_PROVIDER_ACCOUNT_ID_SQL = `SELECT * FROM authjs_accounts WHERE provider = ? AND providerAccountId = ?`;
+export const DELETE_ACCOUNT_BY_PROVIDER_AND_PROVIDER_ACCOUNT_ID_SQL = `DELETE FROM authjs_accounts WHERE provider = ? AND providerAccountId = ?`;
+export const DELETE_ACCOUNT_BY_USER_ID_SQL = `DELETE FROM authjs_accounts WHERE userId = ?`;
+
+// VERIFICATION_TOKEN
+export const GET_VERIFICATION_TOKEN_BY_IDENTIFIER_AND_TOKEN_SQL= `SELECT * FROM authjs_verification_tokens WHERE identifier = ? AND token = ?`;
+export const CREATE_VERIFICATION_TOKEN_SQL = `INSERT INTO authjs_verification_tokens (identifier, expires, token) VALUES (?,?,?)`;
+export const DELETE_VERIFICATION_TOKEN_SQL = `DELETE FROM authjs_verification_tokens WHERE identifier = ? and token = ?`;
+
+// helper functions
+
+// isDate is borrowed from the supabase adapter, graciously
+// depending on error messages ("Invalid Date") is always precarious, but probably fine for a built in native like Date
+function isDate(date: any) {
+  return (
+    new Date(date).toString() !== "Invalid Date" && !isNaN(Date.parse(date))
+  )
+}
+
+// format is borrowed from the supabase adapter, graciously
+function format<T>(obj: Record<string, any>): T {
+  for (const [key, value] of Object.entries(obj)) {
+    if (value === null) {
+      // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+      delete obj[key];
+    }
+
+    if (isDate(value)) {
+      obj[key] = new Date(value);
+    }
+  }
+
+  return obj as T
+}
+
+// D1 doesnt like undefined, it wants null when calling bind
+function cleanBindings(bindings: any[]) {
+  return bindings.map(e=> e === undefined ? null : e)
+}
+
+export async function createRecord<RecordType>(db:BothDB, CREATE_SQL:string, bindings: any[], GET_SQL:string, getBindings: any[]):Promise<RecordType | null> {
+  try {
+    bindings = cleanBindings(bindings)
+    await db.prepare(CREATE_SQL).bind(...bindings).run()
+    return await getRecord<RecordType>(db, GET_SQL, getBindings)
+  } catch(e:any) {
+    console.log(`Error creating record with ${CREATE_SQL} and bindings ${bindings}`)
```

Comment:
Can we remove this line? this would leak the DB Query to the console, so we better not doing it
Please also remove the same line in other helper methods.